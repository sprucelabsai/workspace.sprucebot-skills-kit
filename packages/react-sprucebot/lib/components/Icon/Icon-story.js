"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _react2 = require("@storybook/react");

var _react3 = require("@storybook/addon-knobs/react");

var _Icon = _interopRequireDefault(require("./Icon"));

var stories = (0, _react2.storiesOf)('Icon', module);
stories.addDecorator(_react3.withKnobs);
stories.add('Icon', function () {
  return _react.default.createElement(_Icon.default, {
    icon: (0, _react3.text)('Icon', 'edit'),
    className: (0, _classnames.default)((0, _react3.text)('Class', ''), {
      'u-icon__no-fill u-icon__stroke': (0, _react3.boolean)('Is Line Icon', true)
    })
  });
});