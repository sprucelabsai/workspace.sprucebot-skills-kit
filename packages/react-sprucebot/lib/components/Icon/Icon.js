'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var I = _styledComponents2.default.i.attrs({
	className: function className(_ref) {
		var _className = _ref.className;
		return 'Icon icon ' + (_className || '');
	}
}).withConfig({
	displayName: 'Icon__I',
	componentId: 'ebwesg-0'
})(['font-family:\'Material Icons\';font-weight:normal;font-style:normal;font-size:1em;display:inline-block;height:1em;line-height:1;text-transform:none;letter-spacing:normal;word-wrap:normal;white-space:nowrap;width:1em;direction:ltr;-webkit-font-smoothing:antialiased;text-rendering:optimizeLegibility;-moz-osx-font-smoothing:grayscale;font-feature-settings:\'liga\';']);

var Icon = function Icon(_ref2) {
	var children = _ref2.children,
	    props = _objectWithoutProperties(_ref2, ['children']);

	return _react2.default.createElement(
		I,
		props,
		children
	);
};

Icon.propTypes = {
	color: _propTypes2.default.string
};

exports.default = Icon;