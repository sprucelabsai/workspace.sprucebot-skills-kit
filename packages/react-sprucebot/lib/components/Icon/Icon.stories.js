'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _Icon = require('./Icon');

var _Icon2 = _interopRequireDefault(_Icon);

var _Icon3 = require('./Icon.md');

var _Icon4 = _interopRequireDefault(_Icon3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('Icon', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_Icon4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_Icon2.default,
		null,
		'edit'
	);
})));