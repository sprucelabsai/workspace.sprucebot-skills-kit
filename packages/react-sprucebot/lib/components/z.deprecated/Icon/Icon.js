"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var Icon = function Icon(_ref) {
  var children = _ref.children,
      className = _ref.className,
      props = (0, _objectWithoutProperties2.default)(_ref, ["children", "className"]);
  return _react.default.createElement("i", (0, _extends2.default)({
    className: "Icon icon ".concat(className || '')
  }, props), children);
};

var _default = Icon;
exports.default = _default;