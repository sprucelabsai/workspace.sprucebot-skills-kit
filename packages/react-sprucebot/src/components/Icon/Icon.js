import styled from 'styled-components'
import PropTypes from 'prop-types'
import React from 'react'

const I = styled.i.attrs({
	className: ({ className }) => `Icon icon ${className || ''}`
})`
	font-family: 'Material Icons';
	font-weight: normal;
	font-style: normal;
	font-size: 1em;
	display: inline-block;
	height: 1em;
	line-height: 1;
	text-transform: none;
	letter-spacing: normal;
	word-wrap: normal;
	white-space: nowrap;
	width: 1em;
	direction: ltr;
	/* Support for all WebKit browsers. */
	-webkit-font-smoothing: antialiased;
	/* Support for Safari and Chrome. */
	text-rendering: optimizeLegibility;
	/* Support for Firefox. */
	-moz-osx-font-smoothing: grayscale;
	/* Support for IE. */
	font-feature-settings: 'liga';
`

const Icon = ({ children, ...props }) => <I {...props}>{children}</I>

Icon.propTypes = {
	color: PropTypes.string
}

export default Icon
