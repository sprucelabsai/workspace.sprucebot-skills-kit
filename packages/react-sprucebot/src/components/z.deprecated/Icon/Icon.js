import React from 'react'

const Icon = ({ children, className, ...props }) => (
	<i className={`Icon icon ${className || ''}`} {...props}>
		{children}
	</i>
)

export default Icon
