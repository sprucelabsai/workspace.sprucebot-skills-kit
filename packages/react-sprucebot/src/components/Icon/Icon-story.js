// @flow
import React from 'react'
import cx from 'classnames'
import { storiesOf } from '@storybook/react'
import { withKnobs, text, boolean } from '@storybook/addon-knobs/react'
import Icon from './Icon'

const stories = storiesOf('Icon', module)

stories.addDecorator(withKnobs)

stories.add('Icon', () => (
	<Icon
		icon={text('Icon', 'edit')}
		className={cx(text('Class', ''), {
			'u-icon__no-fill u-icon__stroke': boolean('Is Line Icon', true)
		})}
	/>
))
