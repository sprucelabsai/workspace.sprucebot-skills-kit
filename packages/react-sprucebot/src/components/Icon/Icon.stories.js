import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import Icon from './Icon'
import readme from './Icon.md'

const stories = storiesOf('Icon', module)
stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(readme, withInfo()(() => <Icon>edit</Icon>))
)
