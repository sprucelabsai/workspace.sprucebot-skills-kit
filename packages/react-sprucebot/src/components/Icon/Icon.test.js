import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Icon from './Icon'

test('it renders', () => {
	const tree = renderer.create(<Icon />).toJSON()
	expect(tree).toMatchSnapshot()
})
