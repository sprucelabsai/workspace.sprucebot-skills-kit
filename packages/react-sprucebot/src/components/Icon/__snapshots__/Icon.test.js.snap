// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<i
  className="Icon__I-s1r0bu8t-0 Icon icon  fQySdo"
/>
`;
